### PR TITLE
Improve infinite scroll triggering

### PR DESCRIPTION
**Pull request by ghost**
Infinite scroll triggered relative to last element top rather than window bottom

### CODE REVIEW

File: app/assets/javascripts/app/views/infinite_stream_view.js [hunk 1]
```diff
@@ -95,12 +95,11 @@ app.views.InfScroll = app.views.Base.extend({
   },
 
   infScroll : function() {
-    var $window = $(window)
-      , distFromTop = $window.height() + $window.scrollTop()
-      , distFromBottom = $(document).height() - distFromTop
-      , bufferPx = 500;
+    var $window = $(window),
+        elementTop = this.$el.children().last().offset().top,
+        elementDistance = elementTop - $window.scrollTop() - 500;
```
Comment (svbergerem) [inline app/assets/javascripts/app/views/infinite_stream_view.js]: Since `$window` is used only once I think you can remove the variable.

**Comment by svbergerem** [comment]
This PR should fix #5064.

**Comment by svbergerem** [comment]
Could there be cases where the difference between the top of the last post and the top of the window is more than 500px but you are not able to scroll down anymore? (big screen and small last post?)

Maybe we should keep the current code and just add your code as a second condition for loading more posts?!

**Comment by AugierLe42e** [comment]
The code should execute as soon as the post appears at the bottom of the page. I've tested it on a short last post.

**Comment by svbergerem** [comment]
> The code should execute as soon as the post appears at the bottom of the page.

If the height of your screen is about 500px...

**Comment by AugierLe42e** [comment]
Hmm... I've never really understood how lengths work. >.<
My screen is 1920*1808px. Anyway, give me 5 minutes to change this.

**Comment by svbergerem** [comment]
Merged as 22882aa24d334646d675d71593712fe732ffd380

Thanks!

**Pull request closed without merging**